### PR TITLE
Ai tweaks for sparse galaxy

### PR DESCRIPTION
**Pull request by Dilvish-fo**
Hi guys, long time no see.  I have a little Christmas contribution that I think could help liven up the holidays-- some relatively modest AI tweaks that nonetheless seem to make it a fair bit more competitive, especially in the specific setups this is directed to, which are ones where the monster density is no greater than Low, and where there are at least 40 systems per empire (or 35 if the shape is elliptical).   I prefer to run this at 400 systems and 6 AIs.  

Also, independent of the sparsity, this enables the AI to use the XenoResurrection Lab and the associated extinct species.

I am still getting up to speed on some of the AI changes during my absence, so CJ et al., just let me know if you'd prefer I change the way I tweaked the techs here.

### CODE REVIEW

**Comment by Cjkjvfnby** [comment]
I have exams soon, after that new year + vacation and more exams after.  I am useless until that.  :(

File: default/python/AI/ColonisationAI.py [hunk 1]
```diff
@@ -214,6 +214,9 @@ def survey_universe():
         active_growth_specials.clear()
         if tech_is_complete(TechsListsAI.EXOBOT_TECH_NAME):
             empire_colonizers["SP_EXOBOT"] = []  # get it into colonizer list even if no colony yet
+        for spec_name in AIDependencies.EXTINCT_SPECIES:
+            if empire.buildingTypeAvailable("BLD_COL_"+spec_name):
+                empire_colonizers["SP_"+spec_name] = []  # get it into colonizer list even if no colony yet
```
Comment (Morlic-fo) [inline default/python/AI/ColonisationAI.py:219]: Whitespace around operators.

**Review by Morlic-fo — CHANGES_REQUESTED** [review]
What a nice christmas surprise! Happy to see you again!

**Comment by MatGB** [comment]
I've played two games through with this now and, for the first time in over a year, one of them had to be restarted as the AI overpowered me in the early game: partially due to my own mistakes and partially because it was doing things I wasn't expecting.

I especially like the greater emphasis on detection techs in this, even if it did mess up some of my strategies, but it also made me try out a few new ideas in a way that was fun.

Overall I like the way it feels/plays.

**Comment by Morlic-fo** [comment]
Ok that did not work out as intended. Tried resolving the conflicts using the web interface. Turns out that creates a merge commit.

**Comment by Vezzra** [comment]
> Ok that did not work out as intended. Tried resolving the conflicts using the web interface. Turns out that creates a merge commit.

@Morlic-fo, hopefully fixed. 😃

**Comment by Dilvish-fo** [comment]
Morlic, thanks much for your feedback.  I've made the corrections and fixed the conflict with current master, and was trying to do a bit of testing to confirm things are working OK before pushing the revised branch.   My testing is a little bit slowed down, though, since I ~~can't find any AI logs anymore,...~~ had trouble finding the new location for AI logs (in the .local folder), but I just now found it.

**Comment by dbenage-cx** [comment]
@Dilvish-fo try `~/.local/share/freeorion` for logs (if not, check config file for location of `save-dir`, should be the parent of)

**Comment by Vezzra** [comment]
@Dilvish-fo, do you think you can get this finalized in time for the release?

**Comment by Cjkjvfnby** [comment]
@Dilvish-fo, closing PR by pushing other commit does not looks good for me.   IMHO PR should be closed manually with comment.

**Pull request closed without merging**